src/Core/Mews.Fiscalizations.Core/Compression/GzipCompressionHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.IO.Compression;
+using System.Text;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Mews.Fiscalizations.Core.Compression
+{
+    public static class GzipCompressionHelper
+    {
+        public static string Compress(string inputString)
+        {
+            byte[] buffer = Encoding.UTF8.GetBytes(inputString);
+            var memoryStream = new MemoryStream();
+            using (var gZipStream = new GZipStream(memoryStream, CompressionMode.Compress, true))
+            {
+                gZipStream.Write(buffer, 0, buffer.Length);
+            }
+
+            memoryStream.Position = 0;
+
+            var compressedData = new byte[memoryStream.Length];
+            memoryStream.Read(compressedData, 0, compressedData.Length);
+
+            var gZipBuffer = new byte[compressedData.Length + 4];
+            Buffer.BlockCopy(compressedData, 0, gZipBuffer, 4, compressedData.Length);
+            Buffer.BlockCopy(BitConverter.GetBytes(buffer.Length), 0, gZipBuffer, 0, 4);
+            return Convert.ToBase64String(gZipBuffer);
+        }
+
+        public static string Decompress(string compressedString, Encoding encoding)
+        {
+            byte[] gZipBuffer = Convert.FromBase64String(compressedString);
+            using (var memoryStream = new MemoryStream())
+            {
+                int dataLength = BitConverter.ToInt32(gZipBuffer, 0);
+                memoryStream.Write(gZipBuffer, 4, gZipBuffer.Length - 4);
+
+                var buffer = new byte[dataLength];
+
+                memoryStream.Position = 0;
+                using (var gZipStream = new GZipStream(memoryStream, CompressionMode.Decompress))
+                {
+                    gZipStream.Read(buffer, 0, buffer.Length);
+                }
+
+                return encoding.GetString(buffer);
+            }
+        }
```

Comment:
Same here, how about this

```
        public static string Decompress(byte[] bytes, Encoding encoding)
        {
            using var memoryStream = new MemoryStream(bytes);
            using var outputStream = new MemoryStream();
            using (var decompressStream = new GZipStream(memoryStream, CompressionMode.Decompress))
            {
                decompressStream.CopyTo(outputStream);
            }
            return encoding.GetString(outputStream.ToArray());
        }
```

Reply:
And now since the codes are almost identical, maybe you can make it accept the CompressionMode as a param and have one method instead of 2? 😄 not sure if that's great but consider it

Reply:
The difference is that the both methods takes and return different types. We could abstract parts of the code though, but I don't think it is really worth it.